### ISSUE TITLE
PATCH is allowed as a method, succeeds but doesn't do anything

### ISSUE THREAD

**Issue by gegenschall** [sunscrapers/djoser#251]
I tried using the `PATCH` method to update fields on the user model, i.e. 

```
PATCH /api/auth/me HTTP1.1

{
  "email": "test@example.com"
}
```

The response to this query is then the user object with HTTP status 200 but nothing has changed, i.e. a subsequent GET request returns the old value for `email`.

Either `PATCH` should immediately return 405 or (preferably) it should actually persist the values you pass.

**Comment by piotr-szpetkowski** [comment]
Hi. Few of ideas I have right now:

1. Have you provided `EMAIL_FIELD` or implemented `get_email_field_name` on User model or provided `USER_EMAIL_FIELD_NAME` djoser setting (if you are using Django < 1.11)?
2. Is your email field your `USERNAME_FIELD`?

**Comment by gegenschall** [comment]
`email` is just an example for this behavior. Can be seen with any combination of fields.

I think the bug here is that the `PATCH` method on (at least) the `user` view is allowed but does not do the right thing.

**Comment by piotr-szpetkowski** [comment]
Alright, thanks for the input. I'll look into the issue in more depth this friday.

**Comment by piotr-szpetkowski** [comment]
So, after doing my best trying to replicate the issue in my development environment I wasn't able to do so, however I have few more hints which might lead to the solution:

1. I've noticed that you've tried to use `PATCH /api/auth/me`, which should result in 404. Recently we've updated urlpatterns to require trailing slash for all URLs for sake of consistency - could it be related to the issue you are experiencing?
2. Actually related with the above - which version of djoser are you running?
